lib/puppet/graph/relationship_graph.rb

```diff
@@ -259,11 +259,9 @@ def replace_containers_with_anchors(catalog)
       priority = @prioritizer.priority_of(x)
       add_vertex(admissible[x], priority)
       add_vertex(completed[x], priority)
-    }
-    #
-    # Implement the six requirements listed above
-    #
-    containers.each { |x|
```

Comment:
I've been bitten by the combinable loops cop before, as it doesn't take into account side-effects. For example, previously we walked all of the containers and added verticies for admissible & completed and only after did we add adjacent edges for each container. I'm not 100% certain the order of those operations can be interleaved.

Reply:
Yeah, I think it would be good to err on safety here and remove the cop changes; the `init` code was also slightly changed so that it would search for paths with and without `.sh` differently. This meant the mocking of the test had to change slightly, but I think it was ok to keep in that instance. I will just add some comments to skip this part in the relationship graph.